vars/cloneVSTestTools.groovy

```diff
@@ -0,0 +1,23 @@
+def call(){
+   echo 'Cloning vs test tools to workspace.'
+
+   def organization = getComponentParts()['organization']
+   def branch = env."library.vs-test-tools.version"
+
+   repo = "https://github.com/$organization/niveristand-custom-device-testing-tools"
+   echo "Cloning $repo to $branch directory."
+   testToolsDir = "..\\" + repo.tokenize("/").last()
```

Comment:
This should use the `cloneRepo` function instead of duplicating the code.

Reply:
@buckd I ended up not being able to use the cloneRepo function because it only clones the repo to the branch directory and I need it to clone to the level below the workspace

Here is the code from the cloneRepo function. Notice that it does not include "..\\" nor does it clean up the directory prior to making a new directory (which prevents write errors)

```
   echo "Cloning $repo to workspace."
   cloneDir = repo.tokenize("/").last()
   
   bat "mkdir $cloneDir" 
```
What do you think an alternative is to do this?

Reply:
To add more context here. When you are developing with the test tools, the expectation is that the test tools will be in a folder that is parallel to repo that is utilizing the test tools. It should match the organization layout (org/CD_repo, org/test-tools, org/build-tools). You can see this in the layout below.

![image](https://user-images.githubusercontent.com/33259264/56159763-16da2a00-5f8b-11e9-8dac-ae2912c8b5a2.png)

This is necessary because there is a LabVIEW class dependency between the CD project and VeriStandTestCase.lvclass in the testing tools repo. If the testing tools repo is located somewhere else, then the project cannot find VeriStandTestCase.lvclass and will fail the test.

Using the `cloneRepo` function places the testing tools into the workspace, which is a copy of the repo under test. This is not where the repo under test expects the testing tools, so the test fails.

This change places the niveristand-custom-device-testing-tools into the folder parallel to the workspace folder as you can see in the image below.

![image](https://user-images.githubusercontent.com/33259264/56160203-36258700-5f8c-11e9-87ea-fad4c88f9531.png)

Reply:
This was created as a method of working with the testing tools as a github repo. If it were build into an artifact, then there would be other options available. Such as installing the version of the artifact that corresponds to the version of the CD being tested.

Reply:
I would prefer you modify `cloneRepo.groovy` to handle an optional destination. This prevents code duplication. Something like the following.

```
// cloneRepo.groovy
def call(repo, branch, destination = '') {
   echo "Cloning $repo."
   cloneDir = destination + repo.tokenize("/").last()

   bat "if exist \"$cloneDir\" rmdir /S /Q \"$cloneDir\""
   bat "mkdir \"$cloneDir\""

   if(!branch  || branch == null){
      branch = 'master'
   }

   dir(cloneDir){
      git url: repo, branch: branch
   }

   return cloneDir
}
```

Then, you can just set up your variables in `cloneVSTestTools.groovy`.